### ISSUE TITLE
False Positive: Suppressions and explicit fall-through explanations are still treated as undocumented fall-through.

### ISSUE THREAD

**Issue by Carlson-JLQ** [checkstyle/checkstyle#19388]
# False Positive: Suppressions and explicit fall-through explanations are still treated as undocumented fall-through.

Version
unknown in repository metadata

## Checker
- Checker id: `FallThroughCheck`
- Checker description: Detects switch case groups that lack a terminating statement (like break, return, throw, or continue) and thus allow execution to "fall through" to the next case, unless a specific comment pattern is present.

## Description of the false positive
`diff/output` marks 2 false positives for `FallThroughCheck`. Across these samples, the checker tends to fire on the outer syntax while missing code-level cues that make the pattern harmless, exempted, or at least much weaker than a real bug. The recurring themes are switch control flow, suppression or annotations.

### `NegCase6.java`
Source: `output/codeql/BreakInSwitchCase/src/main/java/scensct/core/neg/NegCase6.java`

The example is built around a `switch` whose case-group layout is the whole point of the test. The source is explicitly annotated to suppress the warning, so reporting it anyway is noise from the developer's point of view. That matches the benchmark's stated intent: Switch statement suppressed with @SuppressWarnings should not flag any fallthrough.

```java
// Switch statement suppressed with @SuppressWarnings should not flag any fallthrough.
package scensct.core.neg;

public class NegCase6 {
    @SuppressWarnings("fallthrough")
    public static void main(String[] args) {
        int x = 1;
        switch (x) {
            case 1:
                System.out.println("Case 1");
                // No break, but suppressed
            case 2:
                System.out.println("Case 2");
                break;
        }
    }
}
```

### `NegCase2.java`
Source: `output/pmd/ImplicitSwitchFallThrough/src/main/java/scensct/core/neg/NegCase2.java`

The example is built around a `switch` whose control-flow shape is slightly unusual because comments, `break`s, or `default` handling change how a human would read the fall-through. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: A fall-through switch branch that is the last branch in the switch should not be flagged as implicit fall-through.

```java
// A fall-through switch branch that is the last branch in the switch should not be flagged as implicit fall-through.
package scensct.core.neg;

public class NegCase2 {
    public static void main(String[] args) {
        int x = 1;
        switch (x) {
            case 1:
                System.out.println("Case 1");
                break;
            case 2:
                System.out.println("Case 2");
                // No terminating statement, but it's the last branch (default)
                // Fall-through to default is allowed without comment
            default:
                System.out.println("Default");
        }
    }
}
```

## Cause analysis
The rule is documented as follows: Detects switch case groups that lack a terminating statement (like break, return, throw, or continue) and thus allow execution to "fall through" to the next case, unless a specific comment pattern is present.

In this set the problem shows up around switch control flow, suppression or annotations. The implementation appears to lean too heavily on surface indicators and not enough on what the code actually does.

That is why these reports read like over-approximation. The code often retains a suspicious signature or control-flow skeleton, but once you read the body, the benchmark's claimed exemption is visible in the source itself.

For `FallThroughCheck`, the practical improvement would be to model those source-level exemptions more explicitly instead of reporting every syntactic near-match.

## References
None known.

**Comment by romani** [comment]
pelase follow issue temaple
